src/tree.rs

```diff
@@ -714,7 +717,8 @@ impl<T> Tree<T> {
     /// # }
     /// ```
     ///
-    pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F) -> Result<(), NodeIdError>
+    pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F)
+                                      -> Result<(), NodeIdError>
         where B: Ord, F: FnMut(&Node<T>) -> B
     {
```

Comment:
Similarly, in a situation where there's just a `return` value that gets pushed to the next line but there's no `where` clause,  I think it would probably be better to have something like:
```
pub fn sort_children_by_key<B, F>(&mut self, node_id: &NodeId, mut f: F)
    -> Result<(), NodeIdError>
{
```
instead of having the `{` on the same line as the `return` value and having it tabbed over to line up with the function args.